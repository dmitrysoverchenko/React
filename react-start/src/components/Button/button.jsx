import React, { Component } from "react";
import "./Button.scss";
import PropTypes from 'prop-types';

class Button extends Component {
  render() {
    const { text, backgroundColor, clickHandler, cancel } = this.props;

    return (
      <button
        style={{ backgroundColor }}
        onClick={(e) => clickHandler(e)}
        className={"button"}
        data-cancel={cancel}
      >
        {text}
      </button>
    );
  }
}

Button.propTypes = {
  text: PropTypes.string,
  backgroundColor: PropTypes.string,
  clickHandler: PropTypes.func,
  cancel: PropTypes.bool,
};

export default Button;
