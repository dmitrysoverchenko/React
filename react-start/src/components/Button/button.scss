.button {
    border-radius: 5px;
    width: 100px;
    color: #fff;
    border: none;
    padding: 12px 10px;
    margin: 0 5px;
    cursor: pointer;
  }