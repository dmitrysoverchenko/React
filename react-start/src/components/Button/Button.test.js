import React from "react";
import { Button } from "./Button";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

const {click} = userEvent;

describe("<Button/>", () => {}