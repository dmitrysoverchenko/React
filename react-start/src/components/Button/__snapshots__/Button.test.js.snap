// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should made snapshot 1`] = `
<button
  class="button"
  data-testid="buttonTest"
/>
`;

exports[`<Button/> should matches snapshot 1`] = `
<button
  class="button"
  data-testid="buttonTest"
/>
`;
